azurerm/resource_arm_operational_insight_workspace.go

```diff
@@ -0,0 +1,224 @@
+	log.Printf("[INFO] preparing arguments for AzureRM Operational Insight workspace creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	skuName := d.Get("sku")
+	sku, err := getSku(skuName)
+	if err != nil {
+		return err
+	}
+
+	retentionInDays := int32(d.Get("retention_in_days").(int))
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := operationalinsights.Workspace{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandTags(tags),
+		WorkspaceProperties: &operationalinsights.WorkspaceProperties{
+			Sku:             sku,
+			RetentionInDays: &retentionInDays,
+		},
+	}
+
+	cancel := make(chan struct{})
+	workspaceChannel, error := client.CreateOrUpdate(resGroup, name, parameters, cancel)
+	workspace := <-workspaceChannel
+	err = <-error
+	if err != nil {
+		return err
+	}
+	// The cosmos DB read rest api again for getting id. Try remove it.
+	// read, err := client.Get(resGroup, name)
+	// if err != nil {
+	//	return err
+	//}
+
+	//if read.ID == nil {
+	//	return fmt.Errorf("Cannot read Operational Inight Workspace '%s' (resource group %s) ID", name, resGroup)
+	//}
+	d.SetId(*workspace.ID)
+
+	return resourceArmOperationalInsightWorkspaceRead(d, meta)
+
+}
+
+func resourceArmOperationalInsightWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	// I don't understand why we can get the meta data. How the framework set it.
+	client := meta.(*ArmClient).workspacesClient
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["workspaces"]
+
+	resp, err := client.Get(resGroup, name)
+	if err != nil {
+		if responseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on AzureRM Operational Insight workspaces '%s': %s", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", resp.Location)
+	d.Set("resource_group_name", resGroup)
+	d.Set("workspace_id", resp.CustomerID)
+	d.Set("portal_url", resp.PortalURL)
+	d.Set("sku", resp.Sku.Name)
+	d.Set("retention_in_days", resp.RetentionInDays)
+
+	sharedKeys, err := client.GetSharedKeys(resGroup, name)
+	if err != nil {
+		log.Printf("[ERROR] Unable to List Shared keys for Operatinal Insight workspaces %s: %s", name, err)
+	} else {
+		d.Set("primary_shared_key", sharedKeys.PrimarySharedKey)
+		d.Set("secondary_shared_key", sharedKeys.SecondarySharedKey)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+	return nil
+}
+
+func resourceArmOperationalInsightWorkspaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).workspacesClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["workspaces"]
+
+	resp, err := client.Delete(resGroup, name)
+
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
```

Comment:
we've got a helper method which also handles connection drops available in `utils.ResponseWasNotFound(resp.Response)` - can we swap this out for that instead?

Reply:
Done!